### PR TITLE
New codemod: `use-set-literal`

### PR DESCRIPTION
**Pull request by drdavella**
## Overview
*A simple codemod that simplifies set literals*

### CODE REVIEW

File: src/core_codemods/use_set_literal.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            case cst.Name("set"):
+                if self.is_builtin_function(original_node):
+                    match original_node.args:
+                        case [cst.Arg(value=cst.List(elements=elements))]:
+                            # Can't use set literal for empty set
+                            if len(elements) == 0:
+                                self.add_change(original_node, self.CHANGE_DESCRIPTION)
+                                return updated_node.with_changes(args=[])
+
+                            self.add_change(original_node, self.CHANGE_DESCRIPTION)
```
Comment (clavedeluna) [inline src/core_codemods/use_set_literal.py]: can move this line to line 20 so it doesn't get called twice

File: src/core_codemods/use_set_literal.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    def leave_Call(self, original_node: cst.Call, updated_node: cst.Call):
+        match original_node.func:
+            case cst.Name("set"):
+                if self.is_builtin_function(original_node):
+                    match original_node.args:
+                        case [cst.Arg(value=cst.List(elements=elements))]:
+                            # Can't use set literal for empty set
+                            if len(elements) == 0:
+                                self.add_change(original_node, self.CHANGE_DESCRIPTION)
```
Comment (clavedeluna) [inline src/core_codemods/use_set_literal.py]: should also be able to call `self.report_change(original_node)` which I think is nicer but they're kinda dupes of each other

File: tests/codemods/test_use_set_literal.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    def test_simple(self, tmpdir):
+        original_code = """
+        x = set([1, 2, 3])
+        """
+        expected_code = """
+        x = {1, 2, 3}
+        """
+        self.run_and_assert(tmpdir, original_code, expected_code)
+
```
Comment (clavedeluna) [inline tests/codemods/test_use_set_literal.py:17]: worth adding `assert len(self.file_context.codemod_changes) == 1` (or 0) to each test case

File: src/core_codemods/use_set_literal.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class UseSetLiteral(BaseCodemod, NameResolutionMixin):
+    NAME = "use-set-literal"
+    SUMMARY = "Use Set Literals Instead of Sets from Lists"
+    REVIEW_GUIDANCE = ReviewGuidance.MERGE_WITHOUT_REVIEW
+    DESCRIPTION = "Replace sets from lists with set literals"
+    REFERENCES: list = []
+
+    def leave_Call(self, original_node: cst.Call, updated_node: cst.Call):
```
Comment (andrecsilva) [inline src/core_codemods/use_set_literal.py:14]: It's lacking the line include/excludes check.

**Review by andrecsilva — CHANGES_REQUESTED** [review]

**Review by andrecsilva — APPROVED** [review]

**Pull request merged**